### PR TITLE
Unit tests for new class and methods

### PR DESCRIPTION
**Pull request by mbvreddy**
## Description

Please describe the issue or feature and the summary of changes made to fix this. 
Added unit test for following new class or methods added as part of feature.

1. com.autotune.analyzer.adapters.MetricAggregationInfoResultsIntSerializer
2. com.autotune.analyzer.recommendations.Variation
3. com.autotune.analyzer.recommendations.engine.ContainerRecommendationProcessor#getPodCountAggrInfo
4. com.autotune.analyzer.recommendations.utils.RecommendationUtils#getCurrentValue
5. com.autotune.analyzer.recommendations.utils.RecommendationUtils#getCurrentValueForNamespace
6. com.autotune.analyzer.recommendations.term.Terms#checkIfMinDataAvailableForTerm(java.util.Map<java.sql.Timestamp,com.autotune.common.data.result.IntervalResults>, com.autotune.analyzer.recommendations.term.Terms, double)
7. com.autotune.analyzer.recommendations.term.Terms#checkIfMinDataAvailableForTermForNamespace(java.util.Map<java.sql.Timestamp,com.autotune.common.data.result.IntervalResults>, com.autotune.analyzer.recommendations.term.Terms, double)

Output after running tests.
```
[INFO] Running com.autotune.analyzer.recommendations.VariationTest
[INFO] Tests run: 27, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.091 s -- in com.autotune.analyzer.recommendations.VariationTest
[INFO] Running com.autotune.analyzer.recommendations.term.TermsTest
[INFO] Tests run: 38, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.043 s -- in com.autotune.analyzer.recommendations.term.TermsTest
[INFO] Running com.autotune.analyzer.recommendations.utils.RecommendationUtilsTest
[INFO] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.033 s -- in com.autotune.analyzer.recommendations.utils.RecommendationUtilsTest
[INFO] Running com.autotune.analyzer.recommendations.engine.ContainerRecommendationProcessorTest
[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.041 s -- in com.autotune.analyzer.recommendations.engine.ContainerRecommendationProcessorTest
[INFO] Running com.autotune.analyzer.adapters.MetricAggregationInfoResultsIntSerializerTest
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.018 s -- in com.autotune.analyzer.adapters.MetricAggregationInfoResultsIntSerializerTest
```

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [X] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [ ] Followed coding guidelines
- [ ] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

## Summary by Sourcery

Add comprehensive unit test coverage for recommendation terms, utilities, variation model, container pod count aggregation, and metric aggregation serialization.

Tests:
- Add TermsTest to validate minimum data availability checks for container and namespace terms across durations, thresholds, and edge cases.
- Add RecommendationUtilsTest to verify current value computation for container and namespace metrics, including pod count derivation and notification handling.
- Add VariationTest to cover configuration of replicas, resource requests/limits, resources map, JSON (de)serialization, and edge conditions.
- Add ContainerRecommendationProcessorTest to exercise pod count aggregation logic from podCount, CPU usage, and memory usage metrics with precedence and error conditions.
- Add MetricAggregationInfoResultsIntSerializerTest to ensure metric aggregation serialization to integer JSON fields across null, zero, negative, large, and formatted values.

### CODE REVIEW

File: src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Unit tests for MetricAggregationInfoResultsIntSerializer
```
Comment (khansaad) [inline src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java:30]: Should we expand this Javadoc to specify everything this class is testing?

File: src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,318 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 Red Hat, IBM Corporation and others.
```
Comment (khansaad) [inline src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java]: Need to update the year here

File: src/test/java/com/autotune/analyzer/recommendations/engine/ContainerRecommendationProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+
+import java.lang.reflect.Method;
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Unit tests for ContainerRecommendationProcessor#getPodCountAggrInfo method
```
Comment (khansaad) [inline src/test/java/com/autotune/analyzer/recommendations/engine/ContainerRecommendationProcessorTest.java:33]: here as well

File: src/test/java/com/autotune/analyzer/recommendations/utils/RecommendationUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+        RecommendationConfigItem result = RecommendationUtils.getCurrentValueForNamespace(
+                AnalyzerConstants.MetricName.namespaceMemoryRequest, filteredResultsMap, testTimestamp, notifications);
+
+        // Then
+        assertNotNull(result);
+        assertEquals(100000.0, result.getAmount());
+        assertEquals("MiB", result.getFormat());
+        assertTrue(notifications.isEmpty());
+    }
+}
```
Comment (khansaad) [inline src/test/java/com/autotune/analyzer/recommendations/utils/RecommendationUtilsTest.java:601]: Missing EOF

File: src/test/java/com/autotune/analyzer/recommendations/VariationTest.java [hunk 5]
```diff
@@ -0,0 +1,480 @@
...
+        
+        variation1.setReplicas(3);
+        variation2.setReplicas(5);
+
+        // Then
+        assertEquals(3, variation1.getReplicas());
+        assertEquals(5, variation2.getReplicas());
+        assertNotEquals(variation1.getReplicas(), variation2.getReplicas());
+    }
+}
```
Comment (khansaad) [inline src/test/java/com/autotune/analyzer/recommendations/VariationTest.java:480]: Missing EOF

**Review by khansaad — CHANGES_REQUESTED** [review]

File: src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java (hunk 2 above)
Comment (mbvreddy) [reply src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java]: Ok

File: src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java (hunk 1 above)
Comment (mbvreddy) [reply src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java:30]: ok

File: src/test/java/com/autotune/analyzer/recommendations/engine/ContainerRecommendationProcessorTest.java (hunk 3 above)
Comment (mbvreddy) [reply src/test/java/com/autotune/analyzer/recommendations/engine/ContainerRecommendationProcessorTest.java:33]: ok

File: src/test/java/com/autotune/analyzer/recommendations/utils/RecommendationUtilsTest.java (hunk 4 above)
Comment (mbvreddy) [reply src/test/java/com/autotune/analyzer/recommendations/utils/RecommendationUtilsTest.java:601]: It is not mandatory.

File: src/test/java/com/autotune/analyzer/recommendations/VariationTest.java (hunk 5 above)
Comment (mbvreddy) [reply src/test/java/com/autotune/analyzer/recommendations/VariationTest.java:480]: It is not mandatory.

File: src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java (hunk 1 above)
Comment (mbvreddy) [reply src/test/java/com/autotune/analyzer/adapters/MetricAggregationInfoResultsIntSerializerTest.java:30]: Typically for unit tests, what is being tested is denoted by the methods we have for each test. For example, 

testSerializeWithAllFieldsPopulated
testSerializeWithPartialFields
testSerializeWithNegativeValues

Maintaining the list again in javadoc comments doesn't help in my opinion.

File: src/test/java/com/autotune/analyzer/recommendations/engine/ContainerRecommendationProcessorTest.java (hunk 3 above)
Comment (mbvreddy) [reply src/test/java/com/autotune/analyzer/recommendations/engine/ContainerRecommendationProcessorTest.java:33]: Typically for unit tests, what is being tested is denoted by the methods we have for each test. For example,

testGetPodCountAggrInfo_WithPodCountMetric
testGetPodCountAggrInfo_WithCpuUsageMetric

**Review by khansaad — APPROVED** [review]
LGTM
